Add ops `remainder`, `floor`, `ceil`, `clamp`, `sort` and `case`

kasper0406 opened the pull request:
Lifted the `remainder`, `floor`, `ceil`, `clamp`, `sort` and `case` ops from @kentslaney's contribution at https://github.com/kasper0406/stablehlo-coreml/pull/28

I added a few unit tests, and got Gemini to refactor the code and make the tests pass.

kentslaney commented on stablehlo_coreml/converter.py:
```diff
@@ -368,11 +370,110 @@ def calculate_result_index(lhs_idx, rhs_idx, acc):
         new_shape = op.result.type.shape
-        reshape_res = mb.reshape(x=x, shape=new_shape)
+        if len(new_shape) == 0:
+            reshape_res = mb.squeeze(x=x)
```
it is, in fact, unsupported

kasper0406 commented:
Thanks for having a look!

 I sat down and followed the logic manually. It took me quite a while to understand it, so I have ended up, making it in a more linear structure, that I find easier to wrap my head around, even thought the code is a bit longer.

Ideally, the we would have some kind of pattern matching tooling like in MLIR (https://mlir.llvm.org/docs/PatternRewriter/), but that seems like a lot of work for this, so I think current matching approach is alright.

Notice that I have removed your support for And/Or ops in the comparator. If you have any concrete test-cases for those, I would be interested in them. I did not hit the code in any of the test cases I added.

Another thing to note, is that I have changed the code to fail if:
1. Someone is attempting a multi-key sort (due to the non-stable argsort, we can not implement it)
2. If someone does a multi-input sort with `is_stable` set to true in the sort op
